tools/cli/go-whisk-cli/commands/action.go

```diff
@@ -523,10 +522,10 @@ func parseAction(cmd *cobra.Command, args []string) (*whisk.Action, error) {
       }
     }
 
-    // For zip-encoded NodeJS action, the code needs to be base64-encoded.
+    // For zip-encoded actions, the code needs to be base64-encoded.
     // We reach this point if the kind has already be determined. Since the extension is not js,
     // this means the kind was specified explicitly.
-    if ext == ".zip" && (strings.HasPrefix(action.Exec.Kind, "nodejs") || action.Exec.Kind == "blackbox") {
+    if ext == ".zip" {
```

Comment:
See the else block starting on line 497. It seems like we wouldn't need a requirement of specifying a kind flag with .zip extensions based on your changes.